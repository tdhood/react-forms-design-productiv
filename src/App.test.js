import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<App />);
  });

  it("contains expected title", function () {
    const result = render(<App />);
    expect(result.queryByText("Prøductïv")).toBeInTheDocument();
  });

  //FIXME: uncomment when quotes added.
  // it ("rendered quotes app", function () {
  //   const result = render(<App />);
  //   expect(result.queryByText("Click here for an inspirational quøte!")).toBeInTheDocument();
  // });

  it("matches snapshot", function () {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  })
});

