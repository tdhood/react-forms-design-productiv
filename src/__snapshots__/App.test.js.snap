// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app matches snapshot 1`] = `
<div>
  <main
    class="App"
  >
    <header
      class="container-fluid pt-4 pb-1"
    >
      <div
        class="container"
      >
        <h1>
          Prøductïv
        </h1>
        <p
          class="lead"
        >
          The best name in todo list management.
        </p>
      </div>
    </header>
    <section
      class="container mt-4"
    >
      <main
        class="TodoApp"
      >
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div>
              <div
                class="EditableTodo"
              >
                <div
                  class="mb-3"
                >
                  <div
                    class="float-end text-sm-end"
                  >
                    <button
                      class="EditableTodo-toggle btn-link btn btn-sm"
                    >
                      Edit
                    </button>
                    <button
                      class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                    >
                      Del
                    </button>
                  </div>
                  <div
                    class="Todo"
                  >
                    <div>
                      <b>
                        Code!
                      </b>
                       
                      <small>
                        (priority: 
                        2
                        )
                      </small>
                    </div>
                    <div>
                      <small>
                        Write some code
                      </small>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="EditableTodo"
              >
                <div
                  class="mb-3"
                >
                  <div
                    class="float-end text-sm-end"
                  >
                    <button
                      class="EditableTodo-toggle btn-link btn btn-sm"
                    >
                      Edit
                    </button>
                    <button
                      class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                    >
                      Del
                    </button>
                  </div>
                  <div
                    class="Todo"
                  >
                    <div>
                      <b>
                        Make dinner
                      </b>
                       
                      <small>
                        (priority: 
                        1
                        )
                      </small>
                    </div>
                    <div>
                      <small>
                        Cook something healthy
                      </small>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="EditableTodo"
              >
                <div
                  class="mb-3"
                >
                  <div
                    class="float-end text-sm-end"
                  >
                    <button
                      class="EditableTodo-toggle btn-link btn btn-sm"
                    >
                      Edit
                    </button>
                    <button
                      class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                    >
                      Del
                    </button>
                  </div>
                  <div
                    class="Todo"
                  >
                    <div>
                      <b>
                        Go to bed
                      </b>
                       
                      <small>
                        (priority: 
                        3
                        )
                      </small>
                    </div>
                    <div>
                      <small>
                        In bed by 11:15
                      </small>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <section
              class="mb-4"
            >
              <h3>
                Top Todo
              </h3>
              <div
                class="Todo"
              >
                <div>
                  <b>
                    Make dinner
                  </b>
                   
                  <small>
                    (priority: 
                    1
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Cook something healthy
                  </small>
                </div>
              </div>
            </section>
            <section>
              <h3
                class="mb-3"
              >
                Add Nü
              </h3>
              <form
                class="NewTodoForm"
              >
                <div
                  class="mb-3"
                >
                  <input
                    aria-label="Title"
                    class="form-control"
                    id="newTodo-title"
                    name="title"
                    placeholder="Title"
                    value="Title"
                  />
                </div>
                <div
                  class="mb-3"
                >
                  <textarea
                    aria-label="Description"
                    class="form-control"
                    id="newTodo-description"
                    name="description"
                    placeholder="Description"
                  >
                    Description
                  </textarea>
                </div>
                <div
                  class="mb-3 d-flex justify-content-between"
                >
                  <div
                    class="w-75 d-flex justify-content-between"
                  >
                    <label
                      class="d-inline-flex"
                      for="newTodo-priority"
                    >
                      Priority:  
                    </label>
                    <select
                      class="form-control form-control-sm d-inline-flex"
                      id="newTodo-priority"
                      name="priority"
                    >
                      <option
                        value="1"
                      >
                        Ultra-Über
                      </option>
                      <option
                        value="2"
                      >
                        Über
                      </option>
                      <option
                        value="3"
                      >
                        Meh
                      </option>
                    </select>
                  </div>
                  <button
                    class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
                  >
                    Gø!
                  </button>
                </div>
              </form>
            </section>
          </div>
        </div>
      </main>
      <footer
        class="App-footer mt-5 text-muted"
      >
        <small>
          Prødutïv is copyright ©2020 by Flüffy Data Enterprises, Inc.
        </small>
      </footer>
    </section>
  </main>
</div>
`;
